File: problems/2348. Number of Zero-Filled Subarrays/cpp/main.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
		int curr = 0;
		long long total = 0;

		for (int i = 0; i < nums.size(); i++) {
			if (nums[i] == 0)  
				curr++;
			else  
				curr = 0;
			
			total += curr;
		}

		return total;
    }
};

int main() {
	auto s =Solution();
	vector<int> nums;

	nums = {1,3,0,0,2,0,0,4};
	assert(s.zeroFilledSubarray(nums) == 6);

	nums = {0,0,0,2,0,0};
	assert(s.zeroFilledSubarray(nums) == 9);
}

File: problems/2348. Number of Zero-Filled Subarrays/python/main.py
from typing import List


class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        count = 0

        curr = 0
        i = 0
        while i < len(nums):
            if nums[i] == 0:
                curr += 1
                count += curr
            else:
                curr = 0

            i += 1

        return count


if __name__ == "__main__":
    s = Solution()
    assert s.zeroFilledSubarray([1, 3, 0, 0, 2, 0, 0, 4]) == 6
    assert s.zeroFilledSubarray([0, 0, 0, 2, 0, 0]) == 9
